---
layout: issue
title: "setFetchMode(Zend_Db::FETCH_OBJ) breaks Zend_Auth_Adapter_DbTable"
id: ZF-2401
---

ZF-2401: setFetchMode(Zend\_Db::FETCH\_OBJ) breaks Zend\_Auth\_Adapter\_DbTable
-------------------------------------------------------------------------------

 Issue Type: Bug Created: 2008-01-09T12:10:09.000+0000 Last Updated: 2008-12-17T13:43:29.000+0000 Status: Closed Fix version(s): 
 Reporter:  Joó Ádám (ceriak)  Assignee:  Jurrien Stutterheim (norm2782)  Tags: - Zend\_Auth
 
 Related issues: 
 Attachments: 
### Description

Using setFetchMode(Zend\_Db::FETCH\_OBJ) produces the following error upon authentication attempt:

Fatal error: Cannot use object of type stdClass as array in /media/sda8/Programok/Web/ZF/Zend/Auth/Adapter/DbTable.php on line 340

 

 

### Comments

Posted by Yann Gascard (cybermac) on 2008-01-16T12:56:56.000+0000

There are already errors reported about this. The thing is they say it depend on other things to be solved.

Meanwhile just replace

 
        // query for the identity
        try {
            $resultIdentities = $this->_zendDb->fetchAll($dbSelect->__toString());
        } catch (Exception $e) {
            /**
             * @see Zend_Auth_Adapter_Exception
             */
            require_once 'Zend/Auth/Adapter/Exception.php';
            throw new Zend_Auth_Adapter_Exception('The supplied parameters to Zend_Auth_Adapter_DbTable failed to '
                                                . 'produce a valid sql statement, please check table and column names '
                                                . 'for validity.');
        }


By

 
        $oldFetchMode = $this->_zendDb->getFetchMode();
        $this->_zendDb->setFetchMode(Zend_Db::FETCH_ASSOC);
    
        // query for the identity
        try {
            $resultIdentities = $this->_zendDb->fetchAll($dbSelect->__toString());
        } catch (Exception $e) {
            /**
             * @see Zend_Auth_Adapter_Exception
             */
            require_once 'Zend/Auth/Adapter/Exception.php';
            throw new Zend_Auth_Adapter_Exception('The supplied parameters to Zend_Auth_Adapter_DbTable failed to '
                                                . 'produce a valid sql statement, please check table and column names '
                                                . 'for validity.');
        }
        $this->_zendDb->setFetchMode($oldFetchMode);


Just the two lines before the code and the last one.

 

 

Posted by Ralph Schindler (ralph) on 2008-01-23T12:15:36.000+0000

Can you check the version of the adapter that is in the trunk/core as of r7598?

Thanks, Ralph

 

 

Posted by Wil Sinclair (wil) on 2008-03-25T20:40:00.000+0000

Please categorize/fix as needed.

 

 

Posted by Jurrien Stutterheim (norm2782) on 2008-10-08T10:57:28.000+0000

This issue has already been fixed.

 

 

Posted by Wil Sinclair (wil) on 2008-12-17T13:43:29.000+0000

Bookkeeping. Closing and assigning all old unassigned issues. The only unassigned issues should be new and unreviewed.

 

 